### PR TITLE
Readme + cmake

### PR DESCRIPTION
**Pull request by RandyGaul**
Still need to work on example section and the documentation for type definitions and whatnot. I'll work on that after I learn more about the code, and am anticipating you guys might work a little on some of these sections in the meantime.

Here is view of updated readme: https://github.com/RandyGaul/datalibrary

### CODE REVIEW

**Comment by RandyGaul** [comment]
https://github.com/wc-duck/datalibrary/issues

**Comment by RandyGaul** [comment]
https://github.com/wc-duck/datalibrary/issues/114

File: README.md [hunk 1]
```diff
@@ -32,70 +32,26 @@ tool\
...
-
-
-## Parts:
-
-generated headers:
-	c/c++
-
-type library ( tlc ):
-	file used by dl itself to load/store/pack/unpack types.
+## The Worklow of Data Library
```
Comment (lundmark) [inline README.md]: Should be workflow

**Comment by lundmark** [comment]
Awesome work! Really big upgrade on the readme. There's just like you said - some things missing etc. I can try to do a push on it when fredrik has reviewed and eventually merged!

File: README.md [hunk 2]
```diff
@@ -1,103 +1,66 @@
...
+* **load**: read an instance from packed form into memory
+* **store**: write an instance to packed form from memory
+* **pack**: pack txt/json instance to packed format
+* **unpack**: unpack from packed format to txt/json
+* generated C/C++ headers from dltlc
+* **type library (tlc)**: the file format used by Data Library itself to load/store/pack/unpack types.
+* **C-library**: data library itself, all the .cpp files in the src directory.
+* **dltlc**: data library type-library-compiler
+* **dl_pack**: tool to pack/unpack/convert instances to/from JSON.
+* **bindings**: python, lua
```
Comment (wc-duck) [inline README.md]: About the bindings... I'm thinking about removing them, or at least move them to separate repos as they aren't really maintained.

**Pull request merged**